packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/webview_controller_delegate.dart

```diff
@@ -0,0 +1,285 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:webview_flutter_platform_interface/src/v4/navigation_callback_delegate.dart';
+import 'package:webview_flutter_platform_interface/src/v4/types/load_request_params.dart';
+
+import 'types/types.dart';
+import 'webview_platform.dart';
+
+/// Interface for a platform implementation of a web view controller.
+///
+/// Platform implementations should extend this class rather than implement it
+/// as `webview_flutter` does not consider newly added methods to be breaking
+/// changes. Extending this class (using `extends`) ensures that the subclass
+/// will get the default implementation, while platform implementations that
+/// `implements` this interface will be broken by newly added
+/// [WebViewCookieManagerDelegate] methods.
+abstract class WebViewControllerDelegate extends PlatformInterface {
+  /// Creates a new [WebViewControllerDelegate]
+  factory WebViewControllerDelegate(WebViewControllerCreationParams params) {
+    final WebViewControllerDelegate webViewControllerDelegate =
+        WebViewPlatform.instance!.createWebViewControllerDelegate(params);
+    PlatformInterface.verify(webViewControllerDelegate, _token);
+    return webViewControllerDelegate;
+  }
+
+  /// Used by the platform implementation to create a new
+  /// [WebViewControllerDelegate].
+  ///
+  /// Should only be used by platform implementations because they can't extend
+  /// a class that only contains a factory constructor.
+  @protected
+  WebViewControllerDelegate.implementation() : super(token: _token);
```

Comment:
I think this class should include a `final WebViewControllerCreationParams params;` field and pass it to this constructor.

Reply:
Good catch, it should and I added it.